### PR TITLE
RCAL-332 populate meta.ref_file

### PR DESCRIPTION
**Pull request by cshanahan1**
GitHub issue, Closes #

Resolves [RCAL-332](https://jira.stsci.edu/browse/RCAL-332)

**Description**

This PR addresses ...

Checklist
- [ ] Tests

- [ ] Documentation

- [ ] Change log

- [ ] Milestone

- [ ] Label(s)

### CODE REVIEW

File: romancal/stpipe/core.py [hunk 1]
```diff
@@ -53,6 +54,19 @@ def finalize_result(self, model, reference_files_used):
...
                 model.cal_logs.append(_LOG_FORMATTER.format(log_record))
 
+        if len(reference_files_used) > 0:
+            for ref_name, ref_file in reference_files_used:
+                if hasattr(model.meta.ref_file, ref_name):
+                    getattr(model.meta.ref_file, ref_name).name = ref_file
+
+        # this will only run if 'parent' is none, which happens when an individual step is being run
+        # or if self is a RomanPipeline and not a RomanStep.
+        print(self.parent)
```
Comment (PaulHuwe) [inline romancal/stpipe/core.py:64]: Should this line be here?

**Review by PaulHuwe — COMMENTED** [review]
Should this have a test?

File: romancal/stpipe/core.py (hunk 1 above)
Comment (PaulHuwe) [reply romancal/stpipe/core.py]: I removed it.

File: romancal/stpipe/core.py [hunk 2]
```diff
@@ -49,10 +50,25 @@ def finalize_result(self, model, reference_files_used):
...
 
+
+        if len(reference_files_used) > 0:
+            for ref_name, ref_file in reference_files_used:
+                if hasattr(model.meta.ref_file, ref_name):
+                    getattr(model.meta.ref_file, ref_name) = ref_file
+
+        # this will only run if 'parent' is none, which happens when an individual step is being run
+        # or if self is a RomanPipeline and not a RomanStep.
+        print(self.parent)
```
Comment (nden) [inline romancal/stpipe/core.py:66]: remove this print

File: CHANGES.rst [hunk 3]
```diff
@@ -1,6 +1,11 @@
 0.7.2 (unreleased)
 ==================
 
+general
+-------
+
+- populate 'ref_file' section in meta after step is run [#492]
```
Comment (nden) [inline CHANGES.rst:7]:
```suggestion
- Populate 'ref_file' section in meta after step is run. [#492]
```

**Review by nden — APPROVED** [review]
Looks good - two minor comments

**Comment by nden** [comment]
I verified this populates the ref_file structure. I'm seeing a validation error raised from `calibration_software_version` on the `cal` file. I don't think it's related to this PR.

**Pull request merged**